### PR TITLE
feat(dpp): document `$createdAt` and `$updatedAt` validation

### PR DESCRIPTION
**Pull request by jawid-h**
## Issue being fixed or feature implemented

Check that `$updatedAt` property is not equal to `$createdAt` for document replace transition 

## What was done?

- added additional error
- added additional validation

## How Has This Been Tested?

- unit tests

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/rs-dpp/src/errors/consensus/codes.rs [hunk 1]
```diff
@@ -141,6 +141,7 @@ impl ErrorWithCode for StateError {
             Self::DocumentTimestampWindowViolationError { .. } => 4008,
             Self::DuplicateUniqueIndexError { .. } => 4009,
             Self::InvalidDocumentRevisionError { .. } => 4010,
+            Self::DocumentTimestampsAreEqualError(_) => 4025,
```
Comment (shumkov) [inline packages/rs-dpp/src/errors/consensus/codes.rs:144]: Had you taken this one from JS DPP?

**Review by shumkov — APPROVED** [review]
👍

**Review by antouhou — APPROVED** [review]

File: packages/rs-dpp/src/errors/consensus/codes.rs (hunk 1 above)
Comment (shumkov) [reply packages/rs-dpp/src/errors/consensus/codes.rs:144]: We already have DocumentTimestampsMismatchError for this needs

**Comment by shumkov** [comment]
I just wondering why we do validation of document data in the state but not in basic?

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**